frontend/src/metabase/visualizations/lib/apply_axis.js

```diff
@@ -160,16 +154,13 @@ export function applyChartTimeseriesXAxis(
       tickInterval,
       chart.width(),
     );
-    chart.xAxis().ticks(tickInterval.rangeFn, tickInterval.count);
-  } else {
-    chart.xAxis().ticks(0);
   }
 
   // pad the domain slightly to prevent clipping
   xDomain = stretchTimeseriesDomain(xDomain, dataInterval);
 
   // set the x scale
-  chart.x(d3.time.scale.utc().domain(xDomain)); //.nice(d3.time[dataInterval.interval]));
+  chart.x(timeseriesScale(tickInterval, getTimezone(series)).domain(xDomain));
```

Comment:
As mentioned above, there's a chance you could have multiple series with different timezones (via dashboard multiseries). Here's what I'd do:

Change `getTimezone` to take a single series object, depending on where `actual_timezone` is moved to:

```
function getTimezone({ data }) {
  // if we put it on the column:
  return data.cols[0].actual_timezone;
  // if we put it on data:
  return data.actual_timezone;
}
```

In this function, do something like:

```
const timezones = _.uniq(series.map(s => getTimezone(s)))
if (timezones.length > 1) {
  warn(multipleTimezoneWarning({ timezones }))
}
const timezone = timezones[0] || DEFAULT_TIMEZONE;

chart.x(timeseriesScale(tickInterval, timezone).domain(xDomain));
```

Not sure if we need a `DEFAULT_TIMEZONE` but maybe just to be safe?